### PR TITLE
Adds main function

### LINKED ISSUE #4715: Create harvest runner main function for dcatus
## User Story

In order for the harvest runner to process a harvest source, datagov wants to create a `main` function in `harvest.py` responsible for processing a harvest source.

## Acceptance Criteria

[ACs should be clearly demoable/verifiable whenever possible. Try specifying them using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications).]

- [ ] GIVEN harvest.py \
  WHEN `main` is called \
  THEN a dcatus source should be harvested

## Background

[Any helpful contextual notes or links to artifacts/evidence, if needed]

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))

[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
[Any security concerns that might be implicated in the change. "None" is OK, just be explicit here!]

## Sketch
- main function steps
  - parse sys.argv and get the jobid 
  - pull the associated harvest source information from the DB via the jobid 
  - create a `HarvestSource` instance using the derived harvest source information
  - get records changes
  - synchronize records
- get a simple harvest source as a test
- this'll be an integration test
[Notes or a checklist reflecting our understanding of the selected approach]

### PR DESCRIPTION
**Pull request by btylerburton**
# Pull Request

Resolves https://github.com/GSA/data.gov/issues/4715

## About

Creates a main function to use when calling the app from the command line

## PR TASKS

- [X] The actual code changes.
- [x] Tests written and passed.
- [ ] Any changes to docs?
- [ ] Bumped version number in [pyproject.toml](https://github.com/GSA/datagov-harvesting-logic/blob/8078c5b9f015ca7fb8a142e4e4b2e22ad2fd49b1/pyproject.toml#L3) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: harvester/harvest.py [hunk 1]
```diff
@@ -455,3 +456,26 @@ def sync(self) -> None:
...
+
+
+if __name__ == "__main__":
+    import argparse
+    import sys
+
+    parser = argparse.ArgumentParser()
+    try:
+        parser.add_argument("jobId", help="job id for harvest job")
+        args = parser.parse_args()
```
Comment (rshewitt) [inline harvester/harvest.py]: there's an existing argument parse function in [utils](https://github.com/GSA/datagov-harvesting-logic/blob/9527e6d320a250577721288919031cf8b0590532/harvester/utils.py#L22). you would just pass sys.argv[1:] to it

File: Makefile [hunk 2]
```diff
@@ -44,6 +44,9 @@ up: ## Sets up local flask and harvest runner docker environments. harvest runne
 	DATABASE_PORT=5433 docker compose up -d
 	docker compose -p harvest-app up db -d
 
+up-unified: ## For testing when you want a shared db between flask and harvester
```
Comment (rshewitt) [inline Makefile:47]: +1

File: harvester/harvest.py (hunk 1 above)
Comment (rshewitt) [reply harvester/harvest.py]: if that feels too granular we can toss it in favor of doing it this way

**Review by rshewitt — DISMISSED** [review]

File: harvester/harvest.py (hunk 1 above)
Comment (btylerburton) [reply harvester/harvest.py]: found it. thanks!

File: .env [hunk 3]
```diff
@@ -10,3 +10,4 @@ HARVEST_SOURCE_URL=http://localhost:${HARVEST_SOURCE_PORT}
 
 CF_API_URL=https://api.fr.cloud.gov
 LM_RUNNER_APP_GUID=f4ab7f86-bee0-44fd-8806-1dca7f8e215a
+CKAN_API_URL=https://catalog-dev.data.gov
```
Comment (FuhuXia) [inline .env]: why does local docker use catalog-dev url?

Comment (btylerburton) [reply .env]: good point. it shouldn't need that. it was for testing.

Comment (btylerburton) [reply .env]: updated

**Review by FuhuXia — APPROVED** [review]
LGTM

**Pull request merged**